app/src/common/shared/com/igalia/wolvic/crashreporting/CrashReporterService.java

```diff
@@ -60,10 +60,17 @@ public static ArrayList<String> findCrashFiles(@NonNull Context aContext) {
         return files;
     }
 
+    private @NonNull WRuntime.CrashReportIntent getCrashReportFromEngine(String defaultAction) {
+        if (EngineProvider.INSTANCE.isRuntimeCreated())
+            return EngineProvider.INSTANCE.getOrCreateRuntime(getBaseContext()).getCrashReportIntent();
+        Log.e(LOGTAG, "Application crashed during startup, before the engine's runtime had been created.");
+        return new WRuntime.CrashReportIntent(defaultAction, "", "", "");
+    }
+
     @Override
     protected void onHandleWork(@NonNull Intent intent) {
         String action = intent.getAction();
-        WRuntime.CrashReportIntent crash = EngineProvider.INSTANCE.getOrCreateRuntime(getBaseContext()).getCrashReportIntent();
+        WRuntime.CrashReportIntent crash = getCrashReportFromEngine(action);
```

Comment:
What about moving here the if () and early return if the runtime is not created? That would prevent the code from creating ane empty intent

Reply:
I tried it and it indeed solves the problem as well, but thought it was preferably to let the crash report code to complete the function and send the broadcast intent. 

I still don't understand completely how the CrashReportService is implemented, launched from Gecko as a foregroundService, bit it seems there is something broken in our implementation, as it suggests the fact that android throws aForegroundServiceDidNotStartInTimeException a few seconds after terminating the app process. 

I hoped that completing the onHandleWork and sending the broadcast would help, but apparently that's not the case. Hence, no strong opinion on this regard and I'm fine to early return as you suggest.